KAA-1070: Add doxygen target to cmake

ohamov opened the pull request:
The initial purpose of the commit is
moving responsibility of the doxygen documentation
generation to cmake build system.

rasendubi commented on client/client-multi/client-c/CMakeLists.txt:
```diff
@@ -189,3 +189,6 @@ install(TARGETS ${KAA_BUILD_SHARED} ${KAA_BUILD_STATIC} DESTINATION lib)
     include(${CMAKE_CURRENT_SOURCE_DIR}/listfiles/UnitTest.cmake)
 endif()
+
+# Doxygen
```
stupid comment

rasendubi commented on client/client-multi/client-c/pom.xml:
```diff
@@ -91,7 +91,7 @@
                         <exclude>build-*/</exclude>
                         <exclude>Makefile</exclude>
-                        <exclude>Doxyfile</exclude>
+                        <exclude>Modules/doxygen</exclude>
```
You should exclude `Doxyfile` only.

rasendubi commented on client/client-multi/client-c/pom.xml:
```diff
@@ -180,7 +180,7 @@
                         <version>1.0.1</version>
                         <configuration>
-                            <configurationFile>Doxyfile</configurationFile>
+                            <configurationFile>Modules/doxygen/Doxyfile</configurationFile>
```
That won't work, as `Doxyfile` now has non-valid output directory.

rasendubi commented on client/client-multi/client-c/Modules/doxygen/CMakeLists.txt:
```diff
@@ -0,0 +1,44 @@
+   # Add target
+   add_custom_target(doxygen
+     ${DOXYGEN_EXECUTABLE}
+     Doxyfile
```
`${OUTPUT_DIRECTORY}/Doxyfile`

rasendubi commented on client/client-multi/client-c/Modules/doxygen/CMakeLists.txt:
```diff
@@ -0,0 +1,44 @@
+     ${DOXYGEN_EXECUTABLE}
+     Doxyfile
+     WORKING_DIRECTORY
+     ${OUTPUT_DIRECTORY}
```
Remove working directory

rasendubi commented on client/client-multi/client-c/Modules/doxygen/CMakeLists.txt:
```diff
@@ -0,0 +1,44 @@
+   # Build documentation for each directory
+   set(INPUT_DIRECTORY ${KAA_SDK_DIR}/src)
+   set(OUTPUT_DIRECTORY ${KAA_SDK_DIR}/target/apidocs/doxygen)
+   set(HTML_EXTRA_FILES ${KAA_SDK_DIR}/README)
```
You shouldn't set HTML_EXTRA_FILES from there. CMake should only set source directory (`@KAA_SDK_DIR@`) and output directory (`@OUTPUT_DIRECTORY@`)

Commit:
KAA-1070: Add doxygen target to cmake

The initial purpose of the commit is
moving responsibility of the doxygen documentation
generation to cmake build system.

rasendubi commented on client/client-multi/client-c/Modules/doxygen/CMakeLists.txt:
```diff
@@ -0,0 +1,32 @@
+
+if(DOXYGEN_FOUND)
+
+   add_custom_target(doxygen
```
You should create a directory for documentation before calling doxygen as it can't create them without help.

ohamov committed:
KAA-1070: Move the procedure of directory creation to cmake

rasendubi commented on client/client-multi/client-c/Modules/doxygen/CMakeLists.txt:
```diff
@@ -0,0 +1,33 @@
+
+   add_custom_target(doxygen
+     COMMAND ${CMAKE_COMMAND} -E make_directory target/apidocs/doxygen
+     COMMAND ${DOXYGEN_EXECUTABLE}
```
Can you join the lines as the real command is `doxygen Doxyfile`.

rasendubi commented on client/client-multi/client-c/Modules/doxygen/CMakeLists.txt:
```diff
@@ -0,0 +1,33 @@
+
+# This cmake list is resposible for building doxygen documentation.
+
+include (FindDoxygen)
```
Can't `find_package` be used instead?

ohamov committed:
KAA-1070: Fix code review
